open-api/rest-catalog-open-api.yaml

```diff
@@ -2747,6 +2747,54 @@ components:
         uuid:
           type: string
 
+    AzureCredentials:
+      type: object
+      properties:
+        account-name:
+          type: string
+        account-key:
+          type: string
+        token:
+          type: string
+
+    AwsCredentials:
+      type: object
+      required:
```

Comment:
I think we might want to include `session-token` and `expires-at-ms` as required because we don't want static credentials passed through this interface.  That could be the source of leaked credentials if implemented without expiration.

Reply:
Hi, I believe it's acceptable for the rest catalog implementation to return the temporarily generated scoped AK/SK without session token, similar to what's supported in some third-party storage services. The inclusion of a `session-token` seems like an unnecessary requirement to me.

Reply:
The session-token is part of the vended credentials - omitting it can mean that the credentials are incomplete and in turn not accepted.

Reply:
I'm not aware of any solutions that don't require `session-token` for temporary credentials (I think Minio and R2 follow AWS here).  If there are examples, we can consider those (or make a separate credential object in the future), but think we should be very strict here so someone doesn't inadvertently build an insecure solution that hands out static credentials.